### PR TITLE
refac: added request-handler test 

### PR DESCRIPTION
**Pull request by energinet-ajw**
The added test ensures that a command or query also has a handler. 

## Description

The test scans the four main assemblies (WebApi, Application, Domain and Infrastructure).

## References

* #972

### CODE REVIEW

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+namespace Energinet.DataHub.Wholesale.WebApi.IntegrationTests.Application;
+
+public class RequestMatchingHandlerTests
+{
+    /// <summary>
+    /// A request can be either a command or a query.
+    /// </summary>
+    [Theory]
+    [InlineAutoData(typeof(Root))]
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:29]: Where is this `Root`? Is it from the integration test project? If so, why is it in the test?

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+using AutoFixture.Xunit2;
+using Energinet.DataHub.Wholesale.Application.Base;
+using FluentAssertions;
+using MediatR;
+using Xunit;
+
+namespace Energinet.DataHub.Wholesale.WebApi.IntegrationTests.Application;
+
+public class RequestMatchingHandlerTests
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:23]: Isn't there a slight mix-up of usage of the terms request vs command vs query in this class?

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            // Assert
+            handlerTypes.Should().ContainSingle(handlerType => IsHandlerForCommand(handlerType, requestType), $"handler for type {requestType} expected");
+        }
+    }
+
+    private static bool IsIRequest(Type type)
+    {
+        // Ignore these interfaces as they don't have a handler
+        if (type == typeof(ICommand) || type == typeof(ICommand<>) || type == typeof(IQuery<>))
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:55]: Do we have any query stuff yet? Are we sure that we're going to have?
I personally have another design in mind, though not settled on anything yet.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:1]: What about notifications/events?

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+{
+    /// <summary>
+    /// A request can be either a command or a query.
+    /// </summary>
+    [Theory]
+    [InlineAutoData(typeof(Root))]
+    [InlineAutoData(typeof(Wholesale.Application.Root))]
+    [InlineAutoData(typeof(Domain.Root))]
+    [InlineAutoData(typeof(Wholesale.Infrastructure.Root))]
+    public void EachRequest_HasMatchingHandler(Type type)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:33]: What are your thoughts on this suggestion?
```suggestion
    public void EachCommand_HasASingleHandler(Type type)
```

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:1]: One thing at a time. When they arrive, the test will arrive.

Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:1]: I stand corrected. Thought that we already had some.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs (hunk 1 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:29]: Added the whole namespace.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs (hunk 2 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:23]: The comment explains that a request cover both, that is, queries and commands.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs (hunk 3 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:55]: We only have this interface. No queries yet. If we decide on a different solution, I will take it out.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:29]: Hmm, can't see that...

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs (hunk 2 above)
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:23]: Shouldn't `IsHandlerForCommand` be named `IsHandlerForRequest` then?

Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs:23]: Yes. Changed.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs (hunk 1 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/Application/RequestMatchingHandlerTests.cs]: Strange. It should be out there.

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**